src/main/java/org/spongepowered/api/entity/living/player/Player.java

```diff
@@ -69,6 +74,17 @@
  */
 public interface Player extends Humanoid, User, Viewer, ChatTypeMessageReceiver, CommandSource {
 
+    /**
+     * Gets whether this {@link Player player} is local to the game or not. Because there
+     * is the possibility of having a game with running on the game client, there
+     * exists the possibility of having {@link ClientPlayer}s as instances, with a
+     * {@link ClientWorld} is potentially populated by {@link RemoteClientPlayer remote players}
+     * and {@link SinglePlayer the single player entity}.
+     *
+     * @return True if this player is local to the game instance
+     */
+    boolean isLocal();
```

Comment:
does this return true if this is the server player instance for who runs the integrated server

Reply:
Since the local player isn't local to the integrated server (`LocalServer`), no. It's still a separate player in the game. This only returns true for `SinglePlayer` instances.

Reply:
Then we can just check `instanceof SinglePlayer` or `Client.getPlayer().map(p -> p == this).orElse(false)`. This doesn't have a point as you don't get the main client player from a player list or such.

Reply:
@liach has a point. The LocalPlayer of the client is kept to the Client and not stored in any spot that would be mixed with other Players/etc. Instanceof will suffice.